File: fizzbuzz/FizzBuzz.hs
import Data.List

main :: IO ()
main = fizzBuzz

fizzBuzz :: IO ()
fizzBuzz = putStrLn $ intercalate "\n" [fizzBuzzSingle n | n <- [1 .. 100]]

fizzBuzzSingle :: Int -> String
fizzBuzzSingle n
  | n `mod` 15 == 0 = "FizzBuzz"
  | n `mod` 5 == 0 = "Buzz"
  | n `mod` 3 == 0 = "Fizz"
  | otherwise = show n

File: FizzBuzz.rs
fn main() {
    for i in 1.. {
        let mut output = String::new();

        if i % 3 == 0 {
            output.push_str("Fizz");
        } 

        if i % 5 == 0 {
            output.push_str("Buzz");
        }

        else if output == "" {
            output = format!("{}", i);
        }
        
        println!("{}", output);
    }
}